gbpservice/nfp/configurator/agents/generic_config.py

```diff
@@ -0,0 +1,468 @@
+
+from gbpservice.nfp.configurator.agents import agent_base
+from gbpservice.nfp.configurator.lib import (
+                            generic_config_constants as gen_cfg_const)
+from gbpservice.nfp.configurator.lib import constants as common_const
+from oslo_log import log as logging
+from gbpservice.nfp.core import main
+import os
+from gbpservice.nfp.configurator.lib import utils
+from gbpservice.nfp.core import poll as nfp_poll
+LOG = logging.getLogger(__name__)
+
+"""Implements APIs invoked by configurator for processing RPC messages.
+
+RPC client of configurator module receives RPC messages from REST server
+and invokes the API of this class. The instance of this class is registered
+with configurator module using register_service_agent API. Configurator module
+identifies the service agent object based on service type and invokes ones of
+the methods of this class to configure the device.
+
+"""
+
+
+class GenericConfigRpcManager(agent_base.AgentBaseRPCManager):
+    def __init__(self, sc, conf):
+        """Instantiates child and parent class objects.
+
+        Passes the instances of core service controller and oslo configuration
+        to parent instance inorder to provide event enqueue facility for batch
+        processing event.
+
+        :param sc: Service Controller object that is used for interfacing
+        with core service controller.
+        :param conf: Configuration object that is used for configuration
+        parameter access.
+
+        """
+
+        super(GenericConfigRpcManager, self).__init__(sc, conf)
+
+    def configure_interfaces(self, context, kwargs):
```

Comment:
The RPC manager class of all agents reflect the functions of the driver. These will be similar to standard agents RPC APIs. So, we have to keep all these functions. However, the code inside these functions can be put in a common function.